src/AutoUI/Filters/Filters.tsx

```diff
@@ -23,6 +27,100 @@ interface FiltersProps<T> {
 	persistFilters?: boolean;
 }
 
+const removeXNoFilterFromObj = (
+	properties: JSONSchema['properties'],
+	xNoFilter: string[] | boolean | undefined,
+): JSONSchema['properties'] => {
+	if (!properties || !xNoFilter) {
+		return properties;
+	}
+	return Object.keys(properties)
+		.filter(
+			(propKey) => !Array.isArray(xNoFilter) || !xNoFilter.includes(propKey),
+		)
+		.reduce(
+			(subAcc, propKey) => ({
+				...subAcc,
+				[propKey]: properties[propKey],
+			}),
+			{},
+		);
```

Comment:
```suggestion
	if (!properties || !Array.isArray(xNoFilter)) {
		return properties;
	}
	const xNoFilterSet = new Set(xNoFilter);
	return Object.fromEntries(
		Object.keys(properties)
		.filter(
			(propKey) => !xNoFilterSet.has(propKey),
		)
		.map((propKey) => [propKey, properties[propKey]])
	);
```

Reply:
TBH, since we already always passing a new object, this could even be a for-of w/ `delete` statements, saving 2 object creations, but no strong opinion on this.